### PR TITLE
Id token

### PR DESCRIPTION
**Pull request by haaakon**
Checking is inspired by what i wrote in sporing. Might not need the adding of = onto the strings to try making it valid base64. Don't actually remember why we needed to do it, but i remember seeing some strings were not valid unless we manually added them. 

Also added tests for Aud and signature checks

### CODE REVIEW

File: Digipost-Test-Dpost Tests/OAuthTests.swift [hunk 1]
```diff
@@ -131,6 +131,25 @@ class OAuthTests: XCTestCase, LUKeychainErrorHandler {
         })
     }
 
+    func testTokenDifferingNonceThanClient() {
+        let nonce = NSString.randomNumberString()
```
Comment (lillesand) [inline Digipost-Test-Dpost Tests/OAuthTests.swift]: Could we use a more explanatory parameter name here? `let invalidNonce = 'invalidNonce'` or something like that?

File: Digipost.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -847,6 +847,8 @@
 		EC31A6351A07AA780079EAC9 /* UploadMenuViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = EC31A6341A07AA780079EAC9 /* UploadMenuViewController.swift */; };
 		EC31A6391A07AA990079EAC9 /* UploadMenuDataSource.swift in Sources */ = {isa = PBXBuildFile; fileRef = EC31A6381A07AA990079EAC9 /* UploadMenuDataSource.swift */; };
 		EC31A63D1A07AC6F0079EAC9 /* UploadMenuTableViewCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = EC31A63C1A07AC6F0079EAC9 /* UploadMenuTableViewCell.swift */; };
+		EC3533541B0C74B900523419 /* OAuthToken+IDToken.swift in Sources */ = {isa = PBXBuildFile; fileRef = ECFAE2651B0B4CFC0029CAD0 /* OAuthToken+IDToken.swift */; };
```
Comment (lillesand) [inline Digipost.xcodeproj/project.pbxproj:850]: :hankey:

File: Digipost-Test-Dpost Tests/OAuthTests.swift (hunk 1 above)
Comment (haaakon) [inline Digipost-Test-Dpost Tests/OAuthTests.swift]: commit: 685f2148b265801c4a5d6e40a4b3985cd924e3d2

File: Digipost.xcodeproj/project.pbxproj (hunk 2 above)
Comment (haaakon) [inline Digipost.xcodeproj/project.pbxproj:850]: Please tell it to Apple, would love if they did something about it.

Comment (lillesand) [inline Digipost.xcodeproj/project.pbxproj:850]: I'd rather post poos on Guthub ;)

File: Digipost/OAuthToken+IDToken.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    class func isIdTokenValid(idToken : String?, nonce : String) -> Bool {
+        if let actualIDToken = idToken {
+            let idTokenContentArray  = idToken?.componentsSeparatedByString(".")
+            if idTokenContentArray?.count == 2 {
+                if let base64EncodedJson = idTokenContentArray?[1] {
+                    var numberOfCharactersAdded = 0
+                    var error : NSError?
+                    var alteredBase64EncodedJson = base64EncodedJson
+                    var base64Data : NSData?
+                    while base64Data == nil {
```
Comment (lillesand) [inline Digipost/OAuthToken+IDToken.swift:27]: Isn't it better to strip the trailing == from both values than to add it? 

To be honest I don't really understand how this code works. What's the significance of adding up to 6 characters?

Comment (haaakon) [inline Digipost/OAuthToken+IDToken.swift:27]: Probably overkill yes, might there have been an issue with the Sporing client that needed to add that. If there is no need we can just strip the trailing, if thats valid base64 in our case

Comment (lillesand) [inline Digipost/OAuthToken+IDToken.swift:27]: Hmm… as long as it works? I don't understand what we use this code for. Could you explain briefly how it works?

Comment (haaakon) [inline Digipost/OAuthToken+IDToken.swift:27]: As i remember the backend delivered base64 that wouldn't be recognized by the app, so it had to add trailing =, probably not as many as 7, but from 0 to 2 i guess.

Comment (haaakon) [inline Digipost/OAuthToken+IDToken.swift:27]: just did a fix that adds up to two, commit: c1090d2

**Pull request merged**